examples/bayes_llama3/llama3/modules/bayesllama.py

```diff
@@ -0,0 +1,338 @@
+### Code adapted from https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py ###
+
+# coding=utf-8
+# Copyright 2022 EleutherAI and the HuggingFace Inc. team. All rights reserved.
+#
+# This code is based on EleutherAI's GPT-NeoX library and the GPT-NeoX
+# and OPT implementations in this library. It has been modified from its
+# original forms to accommodate minor architectural differences compared
+# to GPT-NeoX and OPT used by the Meta AI team that trained the model.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch LLaMA model."""
+
```

Comment:
Add: 
# This code has been adapted from Meta and Huggingface and inherits the above lisence. 
# The original code can be found here: 
# https://github.com/huggingface/transformers/blob/v4.33.0/src/transformers/models/mpt/modeling_mpt.py